### ISSUE TITLE
Determining strategy for redirecting /vaforms/ while preserving HTML for scraping

### ISSUE THREAD

**Issue by ncksllvn** [department-of-veterans-affairs/va.gov-team#4981]
Related to redirect request https://github.com/department-of-veterans-affairs/va.gov-team/issues/5163

To capture its SEO value, we'd like to redirect `va.gov/vaforms/` to our upcoming modernized version, planned to be at `va.gov/find-forms`.

A redirect for this is easy. However, the Lighthouse API that powers our modernized forms search needs to be able to scrape the HTML of the legacy search at `va.gov/vaforms/`. This puts us a weird position - we need to redirect that page, but we also need its HTML to be accessible in some way for us to scrape.

The easiest way seems to be to check the request method - redirect GET requests but leave POST alone. This way, we can scrape the legacy page's HTML by issuing a POST, while we can get the SEO bump that comes from redirecting GETs.

## Acceptance criteria
- [ ] Visiting /vaforms/ redirects to modernized experience at /find-forms/

**Comment by ncksllvn** [comment]
From https://github.com/department-of-veterans-affairs/va.gov-team/issues/3887#issuecomment-577787680 -

Side note - I'm realizing now that this redirect path must be exact - www.va.gov/vaforms/ - to make sure subdirectories aren't also captured.

But then we need to make sure we capture everything about www.va.gov/vaforms/search_action.asp because of the parameters after a search - /vaforms/search_action.asp?FormNo=&tkey=health&Action=Search.

**Comment by MickinSahni** [comment]
pending an estimation and ready for commitment from @ncksllvn

**Comment by ncksllvn** [comment]
A PR for this was merged - https://github.com/department-of-veterans-affairs/devops/pull/6268.

https://dev.va.gov/vaforms/ now redirects to https://dev.va.gov/find-va-forms-beta/, while allowing to POST to `https://www.va.gov/vaforms/search_action.asp`. However, I just realized that the pages past page 1 lead to `Default.asp`, like `https://www.va.gov/vaforms/Default.asp?CurrentPage=3&orderby=`.

:thinking:

We may need to update this a bit to account for `/vaforms/Default.asp`.

**Comment by MickinSahni** [comment]
Hey Nick, few questions
- how does this relate to #5163? is this ticket a blocker to that one? 
- can you add an estimate to this one? 
- is this required to finish this sprint? trying to understand we might commit to this up-front vs. mid-sprint
@ncksllvn

**Comment by ncksllvn** [comment]
Sorry @MickinSahni, the redirects stuff is already a little wild, and particularly with Find Forms. So this ticket was something I had started a sprint or so prior, and it's weird because I'm pretty dependent on DevOps for it, so I'm kinda using this to report back every so often. We can keep it in the backlog if you prefer.

> how does this relate to #5163? is this ticket a blocker to that one?
Basically, #5163 has very strange criteria for implementing the redirects because our Lighthouse API requires access to the page we intend to redirect. #5163 to me, is implementing the redirects into production. This ticket is defining the strategy we'd use to do that.

> can you add an estimate to this one?
Yep, done

> is this required to finish this sprint? trying to understand how we might've commit to this up-front vs. mid-sprint
Well, I had started coordination with DevOps on this before our last planning, and they just updated me that there are redirects in place on dev for us to test. So, it seems like it jumped to In Progress IMO. 

I don't really feel strongly what we do with this ticket. I just wanted a strategy defined so that when it comes time to launch Find Forms, I don't find myself holding up the launch because I can't figure how to redirect properly.
